dotCMS/src/main/java/com/dotcms/rendering/velocity/services/VTLLoader.java

```diff
@@ -0,0 +1,144 @@
+package com.dotcms.rendering.velocity.services;
+
+import com.dotmarketing.util.Config;
+import com.dotmarketing.util.ConfigUtils;
+import com.dotmarketing.util.Logger;
+import com.dotmarketing.util.UtilMethods;
+
+import java.io.BufferedInputStream;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Files;
+
+import org.apache.velocity.exception.ResourceNotFoundException;
+
+import com.liferay.util.FileUtil;
+
+public class VTLLoader implements DotLoader {
+    private static String velocityCanoncalPath;
+    private static String assetCanoncalPath;
+    private static String assetRealCanoncalPath;
+    private static String VELOCITY_ROOT;
+    private static VTLLoader vtlServices;
+
+    private VTLLoader() {
+        init();
+    }
+
+    public static VTLLoader instance() {
+        if (vtlServices == null) {
+            synchronized (VTLLoader.class) {
+                if (vtlServices == null) {
+                    vtlServices = new VTLLoader();
+                }
+            }
+        }
+        return vtlServices;
+    }
+
+    private void init() {
+        String velocityRootPath = Config.getStringProperty("VELOCITY_ROOT", "/WEB-INF/velocity");
+        if (velocityRootPath.startsWith("/WEB-INF")) {
+            String startPath = velocityRootPath.substring(0, 8);
+            String endPath = velocityRootPath.substring(9, velocityRootPath.length());
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Assign this magic number 9 to a well-named constant, and use the constant instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS109)